reframe/core/modules.py

```diff
@@ -207,6 +208,30 @@ def is_module_loaded(self, name):
     def _is_module_loaded(self, name):
         return self._backend.is_module_loaded(Module(name))
 
+    def load_mapping_from_file(self, module_mapping_file):
+        """Update ``self.module_map`` with a mapping defined in a file."""
+        module_mapping = {}
+        try:
+            with open(module_mapping_file, 'r') as f:
+                for line in f.readlines():
+                    line = re.split(r'\s*#\s*', line.strip())[0]
+                    if not line:
+                        continue
+
+                    try:
+                        key, value = re.split(r'\s*:\s*', line.strip(':'))
+                        values_list = re.findall(
+                            r'(?P<value>[A-Za-z0-9_\-/\+\.]+)', value)
+                        module_mapping.setdefault(key, []).extend(values_list)
+                    except ValueError as e:
+                        raise ConfigError(
+                            'Incorrect format of the'
+                            'module mapping file: %s' % e) from e
+        except OSError as e:
+            raise ConfigError from e
```

Comment:
`OSError`s should be propagated as is. They are not configuration errors. So you don't have to do a try/except here.